src/nupic/regions/PyRegion.py

```diff
@@ -177,6 +177,15 @@ def getOutputElementCount(self, name):
     name: the name of the output
     """
 
+  def getAlgorithm(self):
+    """
+    Returns the underlying algorithm that is performing the computation.
+    This method should be overridden by the region subclass.
```

Comment:
Per Scott's note we should be a bit more clear in the comments that the return type cannot be determined in advance and depends on the Region.